src/crystal/system/file.cr

```diff
@@ -47,6 +47,43 @@ module Crystal::System::File
     m | o
   end
 
+  LOWER_ALPHANUM = "0123456789abcdefghijklmnopqrstuvwxyz".to_slice
+
+  def self.mktemp(prefix : String?, suffix : String?, dir : String, random : ::Random = ::Random::DEFAULT) : {LibC::Int, String}
```

Comment:
Is there a reason the name generation logic here needs to be distinct from `::File.tempname`? That method also includes the process ID and the current time. Could we include them here? Could we get rid of them there?

Reply:
I think I considered the implications for `tempname` that but it's probably better done in a follow up to keep things simple.